editor/modes/visual-editor/block.js

```diff
@@ -49,6 +50,15 @@ function VisualEditorBlock( props ) {
 		}
 	}
 
+	const controls = settings.controls && compact(
+		settings.controls.map( ( control ) => {
+			if ( isString( control ) ) {
+				return wp.blocks.controls.find( ctrl => ctrl.slug === control );
```

Comment:
Likely more performant to shape `controls` as an object keyed by slug to avoid iterating the entire set.